### PR TITLE
1247 add defacement event

### PR DESCRIPTION
**Pull request by mssalvatore**
# What does this PR do?

Adds `DefacementEvent`.
Needed for #1247

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?
* [ ] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] Do all unit tests pass?
* [ ] Do all end-to-end tests pass?
* [ ] Any other testing performed?
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

**Review by cakekoa — APPROVED** [review]

File: monkey/common/tags/attack.py [hunk 1]
```diff
@@ -18,3 +18,6 @@
 CREDENTIALS_FROM_PASSWORD_STORES_T1555_TAG = "attack-t1555"  # credential access
 GATHER_VICTIM_HOST_INFORMATION_T1592_TAG = "attack-t1592"
 ACTIVE_SCANNING_T1595_TAG = "attack-t1595"
+DEFACEMENT_T1491_TAG = "attack-t1491"
+INTERNAL_DEFACEMENT_T1491_001_TAG = "attack-t1491.001"
+EXTERNAL_DEFACEMENT_T1491_002_TAG = "attack-t1491.002"
```
Comment (shreyamalviya) [inline monkey/common/tags/attack.py:23]: If we're adding tags for sub-techniques, we should make a note to update the old ones too.

File: monkey/tests/unit_tests/common/agent_events/test_defacement_event.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    EXTERNAL_DEFACEMENT_T1491_002_TAG,
+    INTERNAL_DEFACEMENT_T1491_001_TAG,
+)
+
+AGENT_ID = UUID("012e7238-7b81-4108-8c7f-0787bc3f3c10")
+TIMESTAMP = 1664371327.4067292
+TAGS = frozenset(
+    {DEFACEMENT_T1491_TAG, INTERNAL_DEFACEMENT_T1491_001_TAG, EXTERNAL_DEFACEMENT_T1491_002_TAG}
+)
+DEFACMENT_TARGET = DefacementEvent.DefacementTarget.INTERNAL
```
Comment (shreyamalviya) [inline monkey/tests/unit_tests/common/agent_events/test_defacement_event.py:17]:
```suggestion
DEFACEMENT_TARGET = DefacementEvent.DefacementTarget.INTERNAL
```

File: monkey/tests/unit_tests/common/agent_events/test_defacement_event.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        "tag-2",
+        "tag-3",
+    }
+
+    assert isinstance(DefacementEvent(**dict_with_tags).tags, frozenset)
+
+
+def test_constructor__extra_fields_forbidden():
+    extra_field_dict = DEFACEMENT_SIMPLE_DICT.copy()
+    extra_field_dict["extra_field"] = 99  # red balloons
```
Comment (shreyamalviya) [inline monkey/tests/unit_tests/common/agent_events/test_defacement_event.py:68]: Red balloons?

**Review by shreyamalviya — APPROVED** [review]
Minor comments

File: monkey/tests/unit_tests/common/agent_events/test_defacement_event.py (hunk 3 above)
Comment (ilija-lazoroski) [reply monkey/tests/unit_tests/common/agent_events/test_defacement_event.py:68]: :balloon: :balloon:

Comment (mssalvatore) [reply monkey/tests/unit_tests/common/agent_events/test_defacement_event.py:68]: https://www.youtube.com/watch?v=hiwgOWo7mDc

**Pull request merged**